pilot/pkg/config/kube/ior/route.go

```diff
@@ -583,8 +585,13 @@ func (r *route) findService(gateway *networking.Gateway) (string, string, error)
 		gwSelector.String(), namespaces)
 }
 
-func getRouteName(namespace, name, actualHost string) string {
-	return fmt.Sprintf("%s-%s-%s", namespace, name, hostHash(actualHost))
+func getRouteName(namespace, name, actualHost, serviceNamespace string) string {
+	routeName := fmt.Sprintf("%s-%s-%s", namespace, name, hostHash(actualHost))
+	routeHost := fmt.Sprintf("%s-%s", routeName, serviceNamespace)
+	if len(routeHost) > istiolabels.DNS1123LabelMaxLength {
+		routeName = routeName[:istiolabels.DNS1123LabelMaxLength-(len(serviceNamespace)+1)]
```

Comment:
This doesn't look right. It checks the length of `routeHost`, but then shortens `routeName`?